firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -182,6 +184,14 @@ private MaybeDocument getDocument(DocumentKey key, List<MutationBatch> inBatches
         remoteDocumentCache.getAllDocumentsMatchingQuery(query);
 
     List<MutationBatch> matchingBatches = mutationQueue.getAllMutationBatchesAffectingQuery(query);
+    // It is possible that a PatchMutation can make a document match a query, even if
```

Comment:
This is a useful comment but it interrupts the flow of this otherwise large method. Consider moving this to be a doc comment on your new method (which doesn't otherwise have comments).

Reply:
Done.